### ISSUE TITLE
Just want to share feedback I have found on reddit that has good key takeaways 

### ISSUE THREAD

**Issue by anton-matosov** [conan-io/conan#1522]
From here: https://www.reddit.com/r/cpp/comments/5v10r1/downsides_to_conanio/

Here is what's good with conan: 
1) Python - Makes it super-easy to write the scripts for the packages.
2) Flexible - It doesn't lock you into to a specific build system on either side. You use visual studio to build? Fine! You use cmake? Fine. You use a standard ass makefile. Fine! You use something very special? Fine! (But you would have to write your own generator - which is pluggable and easy to do). 
3) Works with cross-compiling 
4) Can be used to manage our toolchain (we build our cross-compilers and cmake in conan and deploy it to our build slaves to ensure a consistent build environment). 
5) You can host your own server and don't depend on a centralized controlled repository. This is key for us - we build everything inhouse and want to keep it inhouse.

Here are the major problems I have been running into converting all our dependencies: 
1) MingW support is still not that great. Building things that depend on autotools on windows requires some annoyances. But rumors say that an upcoming version will improve this. 
2) CI integration for large sets of dependencies is not great. We solved this by writting a wrapper around conan that builds the dependencies in the right order etc. This could definitely be improved and be a standard upstream. 
3) Confusion about export / build steps some times. I don't know if this is just me having problems wrapping my head around what's happening some times, but at times it seems like conan doesn't detect changes to the conanfiles correctly even if I export my conanfile. I bet this could be fixed if I had time to file a bug with perfect reproducible steps. But I think it shows that parts of the system is just not mature yet and small niggles like this shows up then and then.

TL;DR: Conan is great, a bit immature but our best bet right now for a consistent way to build dependencies for C/C++.

**Comment by memsharded** [comment]
Hi Anton,

Yes, we were aware of this thread in reddit, and I think your analysis is perfect and aligned with our thoughts and current work:

1. MinGW support: it is true that Visual Studio has been the major compiler in Win, and support and features have been around it, since it was the one most demanded. MinGW has been quite a recent demand, specially now that more OSS users are coming (surprinsingly conan has got more early traction in companies than in OSS). So we keep improving its support, mainly with users contributions and feedback.
2. This is a hot area of work. Don't know if you have been able to read: https://jenkins.io/blog/2017/07/07/jenkins-conan We will keep maturing this and proposing new tools and procedures to improve it.
3. We just released conan 0.25 with new commands, one in special: ``conan create``. It is designed to provide an easier to understand workflow, providing more output messages, and also to remove some annoying edges.

Regarding what you say about conan not detecting changes, of course it could be a bug. Looking forward receiving details if you are able to reproduce it, so we can help fixing it.

> Conan is great, a bit immature but our best bet right now for a consistent way to build dependencies for C/C++.

Thanks very much! Keep giving feedback and contributing, and we will keep doing our best to improve it.

**Comment by memsharded** [comment]
Some updates:

- Regarding CI, maybe this has some new insights: https://www.youtube.com/watch?v=xA9yRX4Mdz0
- We have been continuously improving the commands, local-flow, etc.

So, no inmediate action from us for this issue, I think it can be closed, but re-open or comment as necessary. Thanks so much for your feedback!
